sw/airborne/modules/sensors/mag_lis3mdl.c

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2019 Gautier Hattenberger <gautier.hattenberger@enac.fr>
+ *
+ * This file is part of paparazzi.
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @file modules/sensors/mag_lis3mdl.c
+ *
+ * Module wrapper for ST LIS3MDL magnetometers.
+ */
+
+#include "modules/sensors/mag_lis3mdl.h"
+#include "mcu_periph/uart.h"
+#include "pprzlink/messages.h"
+#include "subsystems/datalink/downlink.h"
+#include "generated/airframe.h"
+
+#ifndef LIS3MDL_CHAN_X
+#define LIS3MDL_CHAN_X 0
+#endif
+#ifndef LIS3MDL_CHAN_Y
+#define LIS3MDL_CHAN_Y 1
+#endif
+#ifndef LIS3MDL_CHAN_Z
+#define LIS3MDL_CHAN_Z 2
+#endif
+#ifndef LIS3MDL_CHAN_X_SIGN
+#define LIS3MDL_CHAN_X_SIGN +
+#endif
+#ifndef LIS3MDL_CHAN_Y_SIGN
+#define LIS3MDL_CHAN_Y_SIGN +
+#endif
+#ifndef LIS3MDL_CHAN_Z_SIGN
+#define LIS3MDL_CHAN_Z_SIGN +
+#endif
+
+#if MODULE_LIS3MDL_UPDATE_AHRS
```

Comment:
Why do we need this `UPDATE_AHRS`? If we can just listen to the correct ABI id instead. Maybe we could add an `ABI_ID_NONE` if you don't wanna listen to any magnetometer.
This could make it more future proof if we wanna combine multiple magnetometers.

Reply:
At the moment we are still accessing the global `imu` structure from mag modules, so only one should be explicitly set to be used in AHRS/INS. Once we have a proper way to handle several sensors of the same type, we can get rid of this. Note that pprzlink v2 introduced a component id for this purpose (regarding the message part) but this is not used at all at the moment.

And `ABI_NONE` is already the binding id for receiving nothing (id 0) :)